### PR TITLE
feat(no-unsupported-features): Update to v21.6.1 of node

### PR DESCRIPTION
**Pull request by scagood**
This is a work in progress... I spent some time going through the node docs for this. Unfortunatly this is very much not automatable currently :cry:

Hopefully its a good base to start from though as we keep the list updated from here.

I still need to:
- [x] Add WebCrypto support.
- [x] Add WebStream support.
- [x] Compare the list with `lib/rules/no-unsupported-features/node-builtins.js` (There are some noticable differences)
- [x] ~Extract the globals from the modules to bind them to the trackMap.~
- [x] ~Add `console` as global~
- [x] ~Add `process` as global~

### CODE REVIEW

**Comment by scagood** [comment]
@aladdin-add Do you think that splitting the node modules up like this makes sense?

In the future I would also like to merge the `lib/rules/no-deprecated-api.js` deprecated definitions into the same core dataset.

**Comment by aladdin-add** [comment]
👍 seems a good direction! just a few thoughts:

* 'node_modules' - the dirname name is a bit confusing. 😄 
* Maybe it can be json files, facilitates the possibility of being generated automatically in the future?

**Comment by scagood** [comment]
It would also be lovely to have a mechanism to detect instances of constructed properties :sad:

Maybe its possible using the ts parser :thinking:

```js
const buffer = Buffer.from('asdasd');

// It'd be nice to mark this as deprecated somehow
buffer.parent
```

File: tests/lib/rules/no-unsupported-features/node-builtins.js [hunk 1]
```diff
@@ -5505,7 +5220,7 @@ new RuleTester({ languageOptions: { sourceType: "module" } }).run(
                             messageId: "unsupported",
                             data: {
                                 name: "vm.Module",
-                                supported: "9.6.0",
+                                supported: "(none yet)",
```
Comment (aladdin-add) [inline tests/lib/rules/no-unsupported-features/node-builtins.js:5223]: What's the meaning of "not yet"? experimental?

Comment (scagood) [reply tests/lib/rules/no-unsupported-features/node-builtins.js:5223]: It means that its not stable in node yet :)
https://nodejs.org/api/vm.html#class-vmmodule

Comment (aladdin-add) [reply tests/lib/rules/no-unsupported-features/node-builtins.js:5223]: so the final message is "The "vm.Module" is not supported until Node.js (not yet). The configured version range is '{{version}}'."

https://github.com/eslint-community/eslint-plugin-n/blob/4b12cdc87f7f8daf7045d9b5ff72cbb33bf67d49/lib/util/check-unsupported-builtins.js#L115

maybe we could add a new messageId for this?

Comment (scagood) [reply tests/lib/rules/no-unsupported-features/node-builtins.js:5223]: mmm, that sounds like a nicer option!

Comment (scagood) [reply tests/lib/rules/no-unsupported-features/node-builtins.js]: Resovled in [`a0159d5` (#180)](https://github.com/eslint-community/eslint-plugin-n/pull/180/commits/a0159d50ef207d74258bcbf0966b0e4c4c21162a)

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks!

File: lib/unsupported-features/node-builtins-modules/events.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+const { READ } = require("@eslint-community/eslint-utils")
+
+/** @type {import('../types.js').SupportVersionTree} */
+const static = {
```
Comment (alex-kinokon) [inline lib/unsupported-features/node-builtins-modules/events.js:4]: `static` is a reserved keyword and will throw in strict mode.

Comment (scagood) [reply lib/unsupported-features/node-builtins-modules/events.js:4]: 😱 Good catch

**Pull request merged**